[[ Navigation Bar ]] Add option to highlight none of the icons

bwmilby opened the pull request:
`set the hilitedItem of widget to 0` will disable the highlight
The selected item index allowed minimum was changed to 0
If the hilitedItem is 0, then an empty string will be returned for the name.

bwmilby committed:
LCG Demo - Add Navigation Bar option to highlight none of the icons

`set the hilitedItem of widget to 0` will disable the highlight
The selected item index allowed minimum was changed to 0
If the hilitedItem is 0, then an empty string will be returned for the name.

bwmilby commented:
This was generated as a demo for LCG, but if the team thinks it is worthwhile, I can add an enhancement request to Bugzilla and flesh out the docs a little bit more.

livecodeali commented:
@bwmilby I think this is definitely worthwhile!

bwmilby commented:
Roger.  I’ll put something in BZ and update the PR.

montegoulding commented on extensions/widgets/navbar/notes/lcg-demo.md:
```diff
@@ -0,0 +1 @@
+#LCG Demo - Add Navigation Bar option to highlight none of the icons
```
@bwmilby this note should be `feature-nohighlight.md` or something like that and not mention the LCG demo.

bwmilby replied:
Will submit another commit this evening.  I will fix this and make sure all of the docs are updated.

bwmilby commented:
Possibly a bug, but I have not been able to figure out how to properly reference another LCB library (com.livecode.library.iconsvg).  It isn't invisible, but it doesn't link.  I've only found one other reference to a LCB library (com.livecode.sort referenced from com.livecode.array - but both are modules), but it doesn't render either.

bwmilby committed:
[[ Navigation Bar ]] Add option to highlight none of the icons

Update documentation references for the new 0 option to have nothing highlighted.
Add ability to set the selected icon name to "" to make nothing highlight.
Updated documentation to indicate that custom icons are possible (reference Icon SVG Library).
Renamed note and expanded to document potential backward compatibility issue (a call to `getNavSelectedItemName` in a previous version of the widget with nothing selected will throw an error).

montegoulding commented:
@bwmilby can you explain your issue a bit more... are you saying you are having an issue with `use`?

bwmilby commented:
Issue is with documentation.  For the icons, I think it would be good to point to the SVG Icon Library as a whole instead of listing all of the handlers.  I tried 3 different ways, but none parse as a link.  I have not looked at the source yet.  Since it doesn’t impact the readability of the docs, I submitted this one the way I did.

bwmilby commented on extensions/widgets/navbar/navbar.lcb:
```diff
@@ -256,7 +258,7 @@ navigation bar, the remaining navigation items have their icons reset
 References: hilitedItemIcons(property), itemLabels(property),
-itemNames(property)
+itemNames(property),  com.livecode.library.iconsvg (library)
 
```
This does not parse as an actual link, but does appear in the list of references.

montegoulding commented:
@livecode-vulcan review ok aa9649b

livecode-vulcan commented:
💙 review by @montegoulding ok aa9649b5440ed2345bffd6d12d01e74120ecbeec

livecode-vulcan commented:
😎 test success aa9649b5440ed2345bffd6d12d01e74120ecbeec

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/540)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/532)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/532)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/532)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/532)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2054)
* try-community-universal-ios-iphoneos12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.1/builds/282)
* try-community-universal-ios-iphonesimulator12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.1/builds/281)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2086)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2076)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2074)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2109)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2046)
